libs/base/Data/Fin.idr

```diff
@@ -184,11 +184,17 @@ natToFinLT Z {prf = LTESucc _} = FZ
 natToFinLT (S k) {prf = LTESucc _} = FS $ natToFinLT k
 
 public export
-natToFinLt : (x : Nat) -> {n : Nat} ->
+0 natToFinLemma : (m,n : Nat) -> So (m < n) => LT m n
+natToFinLemma 0 (S k)     = LTESucc LTEZero
```

Comment:
Since it needed to get public exported anyway and it seems at least plausibly of more general utility, how about a _slightly_ less cryptic name like `ltIsLT` (only less cryptic by an inch, I admit).

Reply:
On second thought, maybe not to my suggestion since this lemma is basically the existing `ltReflectsLT` from `Data.Nat`.

Reply:
OK, I moved the lemma to `Data.Nat`, giving it another name and dropping the `So`, because this makes the proof more general (in my opinion). I could of course reintroduce to `So` if that's preferred.

Reply:
I reverted this to using `So` because otherwise this would have been an unnecessary breaking change on `Data.Fin.fromInteger`. It already broke one or more tests in the test suite...